app/javascript/user-recent-reports.vue

```diff
@@ -0,0 +1,62 @@
+<template lang="pug">
+.thread-list.a-card
+  .thread-list__items.has-no-author-image
+    report(
+      v-for='report in reports',
+      :key='report.id',
+      :report='report',
+      :current-user-id='currentUserId'
+    )
+</template>
+
+<script>
+import Report from './report.vue'
+export default {
+  components: {
+    report: Report
+  },
+  props: {
+    userID: { type: String, required: true }
+  },
+  data: function () {
+    return {
+      reports: null,
+      currentUserId: null
+    }
+  },
+  computed: {
+    reportsAPI() {
+      return `/api/users/${this.userID}/recent_reports.json`
+    }
+  },
+  created() {
+    window.onpopstate = () => {
+      this.getReports()
+    }
+    this.getReports()
+  },
+  methods: {
+    getReports() {
+      fetch(this.reportsAPI, {
+        method: 'GET',
+        headers: { 'X-Requested-With': 'XMLHttpRequest' },
+        credentials: 'same-origin',
+        redirect: 'manual'
+      })
+        .then((response) => {
+          return response.json()
+        })
+        .then((json) => {
+          this.reports = []
+          json.reports.forEach((r) => {
+            this.reports.push(r)
+          })
+          this.currentUserId = json.currentUserId
+        })
```

Comment:
JavaScriptはあまり詳しくないので、コードを読んで気になった点を質問させてください！

- なぜ `then` を2回使っているのか
- `forEach`, `push` を使っている理由

```suggestion
        .then((response) => {
          const json = response.json()
          this.reports = json.reports
          this.currentUserId = json.currentUserId
        })
```

`then` を1つにして、 `reports` もそのまま代入できそうかなと思ったので。

Reply:
@sinsoku 
返信が遅くなって申し訳ありません。
- なぜ then を2回使っているのか 
  説明が合っているかわからないのですが、fetchメソッドはPromiseを返し、PromiseはResponseオブジェクトとして解決されるためです。
- forEach, push を使っている理由
  こちらは参考にしたPRのコードをそのまま真似したため、簡潔に書きなおしました。

確認お願いいたします。

Reply:
@harukus 
thenは1つにしても動きそうかなと思ったのですが、これだと動かなくなったりするでしょうか？

```javascript
.then((response) => {
  const json = response.json()
  this.reports = json.reports
  this.currentUserId = json.currentUserId
})
```

コードが短くなるので、動くなら短いコードの方が良いなと思いました！

Reply:
あれ、動かないのか。
それなら今のコードのままで良さそうです！

Reply:
@sinsoku 
ありがとうございます！